rust/main/chains/hyperlane-ethereum/src/rpc_clients/fallback.rs

```diff
@@ -159,6 +166,12 @@ where
 
                 return Ok(value);
             }
+
+            // if we are here, it means that all providers failed
+            // if one of the errors was non-retryable, we stop doing retrying attempts
+            if let Some(e) = non_retriable {
+                return Err(e.into());
```

Comment:
should we still collect all the errors (not just the first one) and return `Err(FallbackError::AllProvidersFailed(errors).into())` here too?

so we could gather the non_retriable errors into a separate array, and if that array is nonempty at this point, we `break`

Then at the end of the funciton, instead of returning `Err(FallbackError::AllProvidersFailed(errors).into())` we return something like
```rust
let all_errors = retryable_errors.extend(nonretryable_errors);
Err(FallbackError::AllProvidersFailed(all_errors).into())
```